app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if session[:id].nil?
+      erb :'users/signup'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if (params[:username].empty? || params[:password].empty? || params[:email].empty?)
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.